ark-core/src/vtxo.rs

```diff
@@ -320,3 +358,294 @@ where
+    /// The original scripts that form the taproot tree
+    scripts: Vec<ScriptBuf>,
+    /// The taproot spend info containing the merkle tree and keys
+    spend_info: TaprootSpendInfo,
+    /// The tweaked public key
+    tweaked_public_key: bitcoin::key::TweakedPublicKey,
+}
+
+impl VirtualUtxoScript {
+    /// Create a new VirtualUtxoScript from a collection of scripts
+    pub fn new<C>(secp: &Secp256k1<C>, scripts: Vec<ScriptBuf>) -> Result<Self, Error>
+    where
+        C: Verification,
+    {
+        if scripts.is_empty() {
+            return Err(Error::ad_hoc("scripts cannot be empty"));
+        }
+
+        let unspendable_key: PublicKey = UNSPENDABLE_KEY.parse().expect("valid key");
+        let (unspendable_key, _) = unspendable_key.inner.x_only_public_key();
+
+        let mut builder = TaprootBuilder::new();
+        if scripts.len() == 1 {
+            // Single script case
+            builder = builder
+                .add_leaf(0, scripts[0].clone())
+                .map_err(|e| Error::ad_hoc(format!("failed to add leaf: {e:?}")))?;
+        } else {
+            // Multiple scripts case - use balanced tree
+            for script in &scripts {
+                builder = builder
+                    .add_leaf(1, script.clone())
+                    .map_err(|e| Error::ad_hoc(format!("failed to add leaf: {e:?}")))?;
+            }
+        }
+
+        let spend_info = builder
+            .finalize(secp, unspendable_key)
+            .map_err(|e| Error::ad_hoc(format!("failed to finalize taproot tree: {e:?}")))?;
+
+        let tweaked_public_key = spend_info.output_key();
+
+        Ok(Self {
+            scripts,
+            spend_info,
+            tweaked_public_key,
+        })
+    }
+
+    /// Decode from a vector of hex-encoded script strings
+    pub fn decode<C>(secp: &Secp256k1<C>, scripts: Vec<String>) -> Result<Self, Error>
+    where
+        C: Verification,
+    {
+        let decoded_scripts: Result<Vec<ScriptBuf>, _> = scripts
+            .into_iter()
+            .map(|hex_script| {
+                let bytes = hex::decode(hex_script)
+                    .map_err(|e| Error::ad_hoc(format!("failed to decode hex: {e}")))?;
+                Ok(ScriptBuf::from_bytes(bytes))
+            })
+            .collect();
+
+        let scripts = decoded_scripts?;
+        Self::new(secp, scripts)
+    }
+
+    /// Encode the scripts to a vector of hex strings
+    pub fn encode(&self) -> Vec<String> {
+        self.scripts
+            .iter()
+            .map(|script| hex::encode(script.as_bytes()))
+            .collect()
+    }
+
+    /// Get the scripts
+    pub fn scripts(&self) -> &[ScriptBuf] {
+        &self.scripts
+    }
+
+    /// Get the spend info
+    pub fn spend_info(&self) -> &TaprootSpendInfo {
+        &self.spend_info
+    }
+
+    /// Get the tweaked public key
+    pub fn tweaked_public_key(&self) -> bitcoin::key::TweakedPublicKey {
+        self.tweaked_public_key
+    }
```

Comment:
Never used.

Reply:
Mh ok I was thinking that this would be good to have at the API level for the people that use the API

Reply:
I think you are right. I was trying to cover everything I noticed, but getters are usually good.